packages/agreement-process/test/testSuspendAgreement.ts

```diff
@@ -0,0 +1,623 @@
+        ...getMockEService(),
+        producerId,
+        descriptors: [descriptor],
+      };
+
+      const agreement: Agreement = {
+        ...getMockAgreement(),
+        consumerId: consumer.id,
+        eserviceId: eservice.id,
+        descriptorId: descriptor.id,
+        producerId: eservice.producerId,
+        state: randomArrayItem(agreementSuspendableStates),
+      };
+
+      await addOneTenant(consumer, tenants);
+      await addOneEService(eservice, eservices);
+      await addOneAgreement(agreement, postgresDB, agreements);
+
+      const requesterId = randomArrayItem([
+        agreement.consumerId,
+        agreement.producerId,
+      ]);
+      const authData = getRandomAuthData(requesterId);
+
+      await agreementService.suspendAgreement(agreement.id, authData, uuidv4());
+
+      const agreementEvent = await readLastAgreementEvent(
+        agreement.id,
+        postgresDB
+      );
+
+      expect(agreementEvent).toMatchObject({
+        type: "AgreementUpdated",
+        event_version: 1,
+        version: "1",
+        stream_id: agreement.id,
+      });
+
+      const actualAgreementSuspended = decodeProtobufPayload({
+        messageType: AgreementUpdatedV1,
+        payload: agreementEvent.data,
+      }).agreement;
+
+      const isConsumer = requesterId === agreement.consumerId;
+      const expectedStamps = {
+        suspensionByConsumer: isConsumer
+          ? {
+              who: authData.userId,
+              when: new Date(),
+            }
+          : agreement.stamps.suspensionByConsumer,
+        suspensionByProducer: !isConsumer
+          ? {
+              who: authData.userId,
+              when: new Date(),
+            }
+          : agreement.stamps.suspensionByProducer,
+      };
+      const expectedSuspensionFlags = {
+        suspendedByConsumer: isConsumer
+          ? true
+          : agreement.suspendedByConsumer ?? false,
+        suspendedByProducer: !isConsumer
+          ? true
+          : agreement.suspendedByProducer ?? false,
+        suspendedByPlatform: false,
+      };
+      const expectedAgreementSuspended: Agreement = {
+        ...agreement,
+        ...expectedSuspensionFlags,
+        state: agreementState.suspended,
+        suspendedAt: agreement.suspendedAt ?? new Date(),
+        stamps: {
+          ...agreement.stamps,
+          ...expectedStamps,
+        },
+      };
+      expect(actualAgreementSuspended).toMatchObject(
+        toAgreementV1(expectedAgreementSuspended)
+      );
+    });
+
+    it("should succeed when requester is Consumer or Producer, Consumer attributes are not satisfied, and the Agreement is in an suspendable state", async () => {
+      /* If the consumer DOES NOT have all the attributes satisfied,
+      the agreement will be suspended with suspendedByPlatform flag set to true
+      and suspendedByConsumer or suspendedByProducer flag set
+      to true depending on the requester (consumer or producer) */
```

Comment:
Could be more useful to have comments like this near the expect statement part? In this way is more readable what the flags meaning, and how change between tests.

Reply:
Makes sense, I moved comments nearer to where I set the flags!